includes/payment-methods/class-wc-stripe-upe-payment-method-cc.php

```diff
@@ -56,4 +54,68 @@ public function get_title( $payment_details = false ) {
 		);
 	}
 
+	/**
+	 * Returns string representing payment method type
+	 * to query to retrieve saved payment methods from Stripe.
+	 */
+	public function get_retrievable_type() {
+		return $this->get_id();
+	}
+
+	/**
+	 * Add payment method to user and return WC payment token.
+	 *
+	 * @param WP_User $user           User to add payment token to.
+	 * @param object  $intent         JSON object for Stripe payment intent.
+	 *
+	 * @return WC_Payment_Token_CC WC object for payment token.
+	 */
+	public function add_token_to_user_from_intent( $user_id, $intent ) {
+		return $this->add_token_to_user_from_payment_method( $user_id, $intent->payment_method );
+	}
+
+	/**
+	 * Add payment method to user and return WC payment token.
+	 *
+	 * This will be used from the WC_Stripe_Payment_Tokens service
+	 * as opposed to WC_Stripe_UPE_Payment_Gateway.
+	 *
+	 * @param string $user_        WP_User ID
+	 * @param object $payment_method Stripe payment method object
+	 *
+	 * @return WC_Payment_Token_CC
+	 */
+	public function add_token_to_user_from_payment_method( $user_id, $payment_method ) {
+		$customer = new WC_Stripe_Customer( $user_id );
+		$token    = $this->create_payment_token_for_user( $user_id, $payment_method );
+
+		$customer->add_payment_method_actions( $token, $payment_method );
+		return $token;
+	}
```

Comment:
This function has been modified slightly to support being called from the Account Add Payment Method screen via the UPE gateway.